File: kbe/res/sdk_templates/client/csharp/kblog/KBELog.cs


namespace KBEngine
{
    using System;
public enum DEBUGLEVEL : int
{
    DEBUG = 0,
    INFO,
    WARNING,
    ERROR,
            
    NOLOG,  // 放在最后面，使用这个时表示不输出任何日志（!!!慎用!!!）
}


public static class KBELog
{
    
    public static DEBUGLEVEL DebugLevel = DEBUGLEVEL.DEBUG;
        
    private static ILogProvider _provider;

    public static void Init(ILogProvider provider)
    {
        _provider = provider;
    }

    public static string GetHead()
    {
        return "[" + DateTime.Now.ToString("yyyy-MM-dd HH:mm:ss fff") + "] ";
    }

    public static void INFO_MSG(string message)
    {
        if (DEBUGLEVEL.INFO >= DebugLevel)
            _provider?.Info(GetHead()+message);
    }

    public static void WARNING_MSG(string message)
    {
        if (DEBUGLEVEL.WARNING >= DebugLevel)
            _provider?.Warn(GetHead()+message);
    }

    public static void ERROR_MSG(string message)
    {
        if (DEBUGLEVEL.ERROR >= DebugLevel)
            _provider?.Error(GetHead()+message);
    }

    public static void DEBUG_MSG(string message)
    {
        if (DEBUGLEVEL.DEBUG >= DebugLevel)
            _provider?.Debug(GetHead()+message);
    }
}
}

File: demo_kbengine_nex_assets/scripts/kbengine_typescript_plugins/KBELog.ts

export default class KBELog
{

    static DEBUG_MSG(msg: string, ...optionalParams: any[]): void
    {
        optionalParams.unshift(msg);
       
        console.debug.apply(this, optionalParams);
        
    }

    static INFO_MSG(msg: string, ...optionalParams: any[]): void
    {
        optionalParams.unshift(msg);
        
        console.info.apply(this, optionalParams);
    }

    static WARNING_MSG(msg: string, ...optionalParams: any[]): void
    {
        optionalParams.unshift(msg);
       
        console.warn.apply(this, optionalParams);
    }

    static ERROR_MSG(msg: string, ...optionalParams: any[]): void
    {
        optionalParams.unshift(msg);
      
        console.error.apply(this, optionalParams);
    }

    static ASSERT(condition?: boolean, message?: string, ...data: any[]): void
    {
        // 使用抛出异常的方式来实现类似断言功能
        if(!condition)
        {
            throw(new Error(message));
        }

        // note：微信小游戏平台不支持，手册中提到的CC_WECHATGAME未定义，无法区分是否微信小游戏平台，
        // console.assert(condition, message, ...data);
    }
}
